app/code/community/Apruve/ApruvePayment/Model/Api/Abstract.php

```diff
@@ -66,47 +168,220 @@ protected function _getApiKey()
      * Check whether payment works in test mode
      * @return bool
      */
-    protected function _getIsTestMode()
+    protected function getIsTestMode()
     {
         return Mage::getStoreConfig('payment/apruvepayment/mode');
     }
 
+    /**
+     * Build Discount Line item
+     * @param Mage_Sales_Model_Quote|Mage_Sales_Model_Order $object
+     * @return array
+     */
+    protected function _getDiscountItem($object)
+    {
+        $helper = Mage::helper('apruvepayment');
+        $discountItem = [];
+        $discountItem['quantity'] = 1;
+        $discountItem['currency'] = $this->getCurrency();
+        $discountItem['description'] = $helper->__('Cart Discount');
+        $discountItem['sku'] = $helper->__('Discount');
+        $discountItem['title'] = $helper->__('Discount');
+
+        if($object instanceof Mage_Sales_Model_Quote) {
+            $discountAmount = $this->convertPrice($object->getBaseSubtotal() - $object->getBaseSubtotalWithDiscount());
+        } elseif($object instanceof Mage_Sales_Model_Order) {
+            $discountAmount = $this->convertPrice($object->getBaseDiscountAmount());
+        } elseif($object instanceof Mage_Sales_Model_Order_Invoice) {
+            $discountAmount = $this->convertPrice($object->getBaseDiscountAmount());
+        } else {
+            return false;
+        }
+        if($discountAmount) {
+            $discountAmount = -1 * abs($discountAmount);
+            $discountItem['price_ea_cents'] = $discountAmount;
+            $discountItem['price_total_cents'] = $discountAmount;
+
+            return $discountItem;
+        } else {
+            return false;
+        }
+    }
+
 
     /**
      * Get Apruve base url based on mode
      * @param bool $secure
      * @return string
      */
-    public function getBaseUrl($secure = false)
+    public function getBaseUrl($secure = true)
     {
         $http = $secure ? 'https://' : 'http://';
-        if($this->_getIsTestMode()) {
-            return $http.'test.apruve.com/';
+        if($this->getIsTestMode()) {
+            return $http . 'test.apruve.com/';
         } else {
-            return $http.'www.apruve.com/';
+            return $http . 'apruve.com/';
+        }
+    }
+
+    /**
+     * Convert price to needed value
+     * As current version supports only USD, convert price to cents
+     * @param float $price
+     * @return float
+     */
+    protected function convertPrice($price)
```

Comment:
Does this convert to the `amount_cents` like value we know and love? If so, shouldn't we do some rounding?